prometheus/vec.go

```diff
@@ -193,10 +186,29 @@ func (m *MetricVec) Delete(labels Labels) bool {
 	if err != nil {
 		return false
 	}
-	if _, ok := m.children[h]; !ok {
+
+	return m.deleteByHash(h, labels)
+}
+
+// deleteByHash removes the metric from the hash bucket h. If there are
+// multiple matches in the bucket, use lvs to select a metric and remove only
+// that metric.
+func (m *MetricVec) deleteByHash(h uint64, values interface{}) bool {
```

Comment:
To maintain correctness and zero-alloc behavior without repeating code, it is necessary. Any errors are caught during test time.

What is the issue? This is private code. Are you just afraid?